doc: in overlays, pull functions from `self`, not `super` (#300416)
NixOS/nixpkgs: doc/using/overlays.chapter.md

  boost = super.boost.override {
    python = self.python3;
  };

>   rr = self.callPackage ./pkgs/rr {
>     stdenv = self.stdenv_32bit;
>   };

Reviewer: ```suggestion
```

Let's avoid the `callPackage` example entirely for now, since it's a bit confusing (both `self.callPackage` and `super.callPackage` use dependencies from `self`).

Author: I left the `callPackage` in originally as a demonstration of pulling a function from `self`. Is there a less confusing example that we can use, perhaps a fetcher (like the `xz` example)?

Reviewer: Yeah a fetcher sounds pretty good :)

}
```

> The first argument (`self`) corresponds to the final package set. You should use this set for the dependencies of all packages specified in your overlay, including any nixpkgs functions. In the example above, `self` provides all the overridden dependencies of `boost` and `rr`, as well as the `callPackage` function.

Reviewer: ```suggestion
The first argument (`self`) corresponds to the final package set. You should use this set for the dependencies of all packages specified in your overlay, including any nixpkgs functions. In the example above, `self` provides the Python dependency of Boost.
```
